### DISCUSSION TITLE
How to disable syncing for deployment with API (trying telepresence)

### DISCUSSION THREAD

**Discussion by ghost** [argoproj/argo-cd#9477] [Q&A]
I am trying to use telepresence, however each time it deploys the sidecar and updates the deploymeny.yaml so that it works, argocd syncs back the deployment to original state, so I am not able to get it working. Is there any way how to disable syncing with API for a particular deployment? So that I could in my script do the following:

1) Disable syncing
2) Telepresence connects and updates deployment, I will do the trobleshooting etc.
3) Enable syncing back after finishing working with telepresence

How would you approach this? I would like to do this in automated way, so that I do not need to login to ArgoCD each time I would like to debug something.

**Comment by ishitasequeira** [comment]
I don't think there is a way to disable syncing only for a specific deployment of an application. Instead, you can disable the automatic syncing of the application itself. You can disable automated sync by updating sync policy of the application yaml to below.

```
  syncPolicy: {}
```

More details about the API documentation that you could use can be found at url `${ARGOCD_CLUSTER}/swagger-ui#operation/ApplicationService_Update`.
